### PR TITLE
feat(clamm): UI features (DOP-517)

### PR DESCRIPTION
**Pull request by Aercwarden**
Features:

1. Strikes chain filter
image

2. LP positions summary with refresh and select all button

image

3. Multi exercise 

image

### CODE REVIEW

**Review by witherblock — APPROVED** [review]

**Comment by witherblock** [comment]
@hibou-de-nuit please review too

File: apps/dapp/src/components/clamm/PositionsTable/components/Positions/BuyPositions/index.tsx [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+        } catch (err) {
+          const error = err as BaseError;
+          console.error(err);
+          toast.error(error.shortMessage);
+        }
+      }
+      toast.remove(loadingToastId);
+    },
+    [selectedOptionsPool, walletClient, updateBuyPositions],
+  );
```
Comment (hibou-de-nuit) [inline apps/dapp/src/components/clamm/PositionsTable/components/Positions/BuyPositions/index.tsx:149]: No need to memoize. The function causes rerender every time the params change.

File: apps/dapp/src/components/clamm/PositionsTable/components/Positions/columnHelpers/tradeHistory.tsx [hunk 2]
```diff
@@ -126,13 +126,13 @@ export const historyPositionsColumns = [
               <span className="text-[13px] flex items-center space-x-[4px]">
                 <span
                   className={cn(
-                    ' text-stieglitz',
+                    'text-white',
```
Comment (hibou-de-nuit) [inline apps/dapp/src/components/clamm/PositionsTable/components/Positions/columnHelpers/tradeHistory.tsx:129]: Don't override like this, use ternary

File: apps/dapp/src/components/clamm/PositionsTable/index.tsx [hunk 3]
```diff
@@ -141,32 +123,33 @@ const PositionsTable = () => {
...
+
+    return () => {
+      document.removeEventListener('visibilitychange', handleVisibilityChange);
+    };
+  }, [updateBuyPositions, updateLPPositions, positionsTypeIndex]);
 
   return (
     <div className="w-full flex-col items-center justify-center space-y-[12px]">
-      <div className="w-full flex flex-row items-center justify-between">
+      <div className="w-full flex flex-row items-center">
```
Comment (hibou-de-nuit) [inline apps/dapp/src/components/clamm/PositionsTable/index.tsx:145]: flex-row is redundant

File: apps/dapp/src/components/clamm/StrikesChain/components/FilterSettings/components/FilterSettingsButton/index.tsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                  >
+                    Apply
+                  </Button>
+                </Close>
+              </div>
+            </div>
+          </Content>
+        </Portal>
+      </Root>
+    </>
```
Comment (hibou-de-nuit) [inline apps/dapp/src/components/clamm/StrikesChain/components/FilterSettings/components/FilterSettingsButton/index.tsx:146]: No need the fragment

File: apps/dapp/src/components/clamm/StrikesChain/components/FilterSettings/components/RangeSelector/components/Slider/LiquidityThresholdInput/index.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
+import React, { ChangeEvent, useCallback, useState } from 'react';
+
+import * as ToggleGroup from '@radix-ui/react-toggle-group';
+
+type Props = {
+  threshold: number[];
+  setThreshold: (v: number[]) => void;
+};
+const LiquidityThresholdInput = ({ threshold, setThreshold }: Props) => {
+  const [type, setType] = useState(threshold[1] === 0 ? 'USD' : 'Options');
```
Comment (hibou-de-nuit) [inline apps/dapp/src/components/clamm/StrikesChain/components/FilterSettings/components/RangeSelector/components/Slider/LiquidityThresholdInput/index.tsx:10]: Client-side and server-side data mismatch errors could arise from setting default state like this.

File: apps/dapp/src/components/clamm/StrikesChain/components/FilterSettings/components/RangeSelector/index.tsx [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+      const strike = payload[0].payload.strike;
+      const totalLiquidity = payload[0].payload.liquidity;
+      const availableLiquidity = payload[1].payload.availableLiquidity;
+      const optionsAvailable = payload[1].payload.optionsAvailable;
+      return (
+        <div className="custom-tooltip flex flex-col text-[10px] border border-carbon w-full h-full bg-umbra p-[4px]">
+          <p className="flex items-center space-x-[4px]">
+            <span className="text-stieglitz">Strike:</span>
+            <span>{formatAmount(strike, 4)}</span>
+          </p>
```
Comment (hibou-de-nuit) [inline apps/dapp/src/components/clamm/StrikesChain/components/FilterSettings/components/RangeSelector/index.tsx:68]: reuse components

File: apps/dapp/src/components/clamm/StrikesChain/components/StrikesTable.tsx [hunk 7]
```diff
@@ -202,7 +175,25 @@ const StrikesTable = () => {
...
+      .filter(({ liquidityAvailableUsd, optionsAvailable }) => {
+        if (filterSettings.liquidityThreshold[1] === 0) {
+          return (
+            filterSettings.liquidityThreshold[0] < Number(liquidityAvailableUsd)
+          );
+        } else {
+          return (
+            filterSettings.liquidityThreshold[0] < Number(optionsAvailable)
+          );
+        }
```
Comment (hibou-de-nuit) [inline apps/dapp/src/components/clamm/StrikesChain/components/StrikesTable.tsx:189]:
```tsx
filterSettings.liquidityThreshold[1] === 0 ? filterSettings.liquidityThreshold[0] < Number(liquidityAvailableUsd) : filterSettings.liquidityThreshold[0] < Number(optionsAvailable)
```

**Review by hibou-de-nuit — CHANGES_REQUESTED** [review]
UX Suggestion:
- Add labels next to the icons on the table for better clarity. It's a little obscure rn

File: apps/dapp/src/components/clamm/PositionsTable/components/Positions/BuyPositions/components/PositionSummary/index.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+          <span>{formatAmount(totalPremiumUsd, 3)}</span>
+        </span>
+      </div>
+      <div className="flex items-center justify-center space-x-[4px]">
+        <span className="text-stieglitz text-xs">Total size:</span>
+        <span className="text-xs flex items-center justify-center space-x-[4px]">
+          <span>{formatAmount(totalOptions, 3)}</span>
+          <span className="text-stieglitz text-xs">{callTokenSymbol}</span>
+        </span>
+      </div>
```
[the remark this answers is not in the document]
Comment (Aercwarden) [reply apps/dapp/src/components/clamm/PositionsTable/components/Positions/BuyPositions/components/PositionSummary/index.tsx:42]: Not required, very specific to use case and won't be reused

File: apps/dapp/src/components/clamm/PositionsTable/components/Positions/BuyPositions/index.tsx [hunk 9]
```diff
@@ -0,0 +1,335 @@
...
+    },
+    [markPrice],
+  );
+
+  const unselectPosition = useCallback((key: number) => {
+    setSelectedPositions((prev) => {
+      prev.delete(key);
+      return new Map(prev);
+    });
+  }, []);
```
[the remark this answers is not in the document]
Comment (Aercwarden) [reply apps/dapp/src/components/clamm/PositionsTable/components/Positions/BuyPositions/index.tsx:57]: it is used as dependency, hence as to be memoized

File: apps/dapp/src/components/clamm/PositionsTable/components/Positions/BuyPositions/index.tsx (hunk 1 above)
Comment (Aercwarden) [reply apps/dapp/src/components/clamm/PositionsTable/components/Positions/BuyPositions/index.tsx:149]: has to change as per parameters

File: apps/dapp/src/components/clamm/StrikesChain/components/FilterSettings/components/RangeSelector/index.tsx (hunk 6 above)
Comment (Aercwarden) [reply apps/dapp/src/components/clamm/StrikesChain/components/FilterSettings/components/RangeSelector/index.tsx:68]: used only once

File: apps/dapp/src/components/clamm/PositionsTable/components/Positions/BuyPositions/index.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (hibou-de-nuit) [reply apps/dapp/src/components/clamm/PositionsTable/components/Positions/BuyPositions/index.tsx:57]: When ```key``` is changed, unselectPosition is recreated during rerender. This in turn updates the dependency array [here](https://github.com/dopex-io/elvarg/blob/1eb412c965a04c0016795267ff99bcc9cf0f1d51/apps/dapp/src/components/clamm/PositionsTable/components/Positions/BuyPositions/index.tsx#L65), which causes that handler to be recreated as well. So there isn't any memoization happening

**Review by hibou-de-nuit — APPROVED** [review]
Refer pending comments. LGTM

**Pull request merged**